scripts/create_channel.go

```diff
@@ -0,0 +1,72 @@
+package scripts
+
+import (
+	"fmt"
+
+	"github.com/mattermost/mattermost-server/v6/model"
+	"go.uber.org/zap"
+
+	"github.com/Brightscout/mattermost-load-test-scripts/serializers"
+	"github.com/Brightscout/mattermost-load-test-scripts/utils"
+)
+
+func CreateChannels(config *serializers.Config, logger *zap.Logger) error {
+	client := model.NewAPIv4Client(config.ConnectionConfiguration.ServerURL)
+	if _, _, err := client.Login(config.ConnectionConfiguration.AdminEmail, config.ConnectionConfiguration.AdminPassword); err != nil {
+		return err
+	}
+
+	var newChannels []*serializers.ChannelResponse
+	response, err := utils.LoadResponse()
+	if err != nil {
+		return err
+	}
+
+	for _, channel := range config.ChannelsConfiguration {
+		team, _, err := client.GetTeamByName(channel.MMTeamName, "")
```

Comment:
Have you accounted the case when the err is not nil and team is nil and below you are trying to access team.id ?

Reply:
discussed